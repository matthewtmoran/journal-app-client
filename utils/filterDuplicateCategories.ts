import ICategory from "../interfaces/ICategory";

const filterDuplicateCategories = (used: ICategory[], unused: ICategory[]) => {
  const hash: { [key: string]: ICategory } = {};

  for (let i = 0; i < used.length; i++) {
    const item = used[i];
    hash[item.name] = item;
  }

  for (let i = 0; i < unused.length; i++) {
    const item = unused[i];
    hash[item.name] = item;
  }

  return Object.keys(hash).map((key) => {
    return hash[key];
  });
};

export default filterDuplicateCategories;
