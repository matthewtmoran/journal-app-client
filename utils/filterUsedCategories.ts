import ICategory from "../interfaces/ICategory";

const filterUsedCategories = (used: ICategory[], all: ICategory[]) => {
  // if it's in used, remove from all

  const filtered = all.filter((cat) => {
    const exists = used.find((c) => {
      return c.name === cat.name;
    });

    return !exists;
  });

  return filtered ? filtered : [];
};

export default filterUsedCategories;
