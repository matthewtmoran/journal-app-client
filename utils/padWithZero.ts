const padWithZero = (number: number) => {
  const string = number.toString();
  if (number < 10) {
    return "0" + string;
  }
  return string;
};

export default padWithZero;
