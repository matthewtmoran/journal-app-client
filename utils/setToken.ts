import AsyncStorage from "@react-native-community/async-storage";

const setToken = async (token: string) => {
  try {
    await AsyncStorage.setItem("@token", token);
  } catch (e) {
    console.log({ e });
    // saving error
  }
};

export default setToken;
