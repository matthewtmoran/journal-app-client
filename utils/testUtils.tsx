import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { createStackNavigator } from "@react-navigation/stack";
import { EntryProvider } from "../state/entry-context";
import { MockedProvider } from "@apollo/react-testing";
import { ENTRIES_QUERY } from "../queries/queries";

const Stack = createStackNavigator();

export function renderWithNavigation(component: any) {
  const App = () => <NavigationContainer>{component}</NavigationContainer>;

  return { ...render(<App />) };
}

export const renderWithApolloProvider = (
  Component: any,
  props: any,
  params = {}
) => {
  return (
    <MockedProvider mocks={{}}>
      <Component {...props} />
    </MockedProvider>
  );
};
