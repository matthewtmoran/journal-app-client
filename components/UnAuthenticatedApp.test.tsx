import React from "react";
import UnAuthentitcatedApp from "./UnAuthenticatedApp";
import { renderWithNavigation } from "../utils/testUtils";

describe("UnAuthentitcatedApp", () => {
  it("Should render welcome screen by default", async () => {
    const { findByText, baseElement } = renderWithNavigation(
      <UnAuthentitcatedApp />
    );
    const welcomeText = await findByText("Welcome To Journal App Î²");
    expect(welcomeText).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
