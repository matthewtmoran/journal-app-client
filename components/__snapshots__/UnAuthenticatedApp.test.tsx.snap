// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnAuthentitcatedApp Should render welcome screen by default 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNCSafeAreaView
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              pointerEvents="box-none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              transitionSpec={
                Object {
                  "close": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "flex": 1,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 1,
                          "width": -1,
                        },
                        "shadowOpacity": 0.3,
                        "shadowRadius": 5,
                        "top": 0,
                        "width": 3,
                      }
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignContent": "center",
                              "alignItems": "center",
                              "display": "flex",
                              "height": "100%",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <View>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "fontSize": 30,
                                    "margin": 20,
                                  },
                                  Object {
                                    "fontFamily": "roboto-light",
                                  },
                                ]
                              }
                            >
                              Welcome To Journal App Î²
                            </Text>
                            <TouchableOpacity
                              activeOpacity={0.2}
                            >
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "#144568",
                                    "borderRadius": 4,
                                    "marginVertical": 8,
                                    "paddingHorizontal": 25,
                                    "paddingVertical": 12,
                                    "width": "100%",
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#fff",
                                        "fontSize": 18,
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "fontFamily": "roboto-light",
                                      },
                                    ]
                                  }
                                >
                                  Log In
                                </Text>
                              </View>
                            </TouchableOpacity>
                            <TouchableOpacity
                              activeOpacity={0.2}
                            >
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "#fff",
                                    "borderRadius": 4,
                                    "marginVertical": 8,
                                    "paddingHorizontal": 25,
                                    "paddingVertical": 12,
                                    "width": "100%",
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#144568",
                                        "fontSize": 18,
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "fontFamily": "roboto-light",
                                      },
                                    ]
                                  }
                                >
                                  Register
                                </Text>
                              </View>
                            </TouchableOpacity>
                          </View>
                        </View>
                      </View>
                      <View
                        pointerEvents="box-none"
                      >
                        <View
                          accessibilityElementsHidden={false}
                          importantForAccessibility="auto"
                          pointerEvents="box-none"
                          style={null}
                        >
                          <View
                            pointerEvents="box-none"
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "zIndex": 0,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "backgroundColor": "rgb(255, 255, 255)",
                                  "borderBottomColor": "rgb(224, 224, 224)",
                                  "flex": 1,
                                  "shadowColor": "rgb(224, 224, 224)",
                                  "shadowOffset": Object {
                                    "height": 0.5,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.85,
                                  "shadowRadius": 0,
                                }
                              }
                            />
                          </View>
                          <View
                            pointerEvents="box-none"
                            style={
                              Object {
                                "height": 64,
                                "maxHeight": undefined,
                                "minHeight": undefined,
                                "opacity": undefined,
                                "transform": undefined,
                              }
                            }
                          >
                            <View
                              pointerEvents="none"
                              style={
                                Object {
                                  "height": 20,
                                }
                              }
                            />
                            <View
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <View
                                pointerEvents="box-none"
                                style={
                                  Object {
                                    "marginHorizontal": 16,
                                  }
                                }
                              >
                                <Text
                                  accessibilityRole="header"
                                  numberOfLines={1}
                                  style={
                                    Object {
                                      "color": "rgb(28, 28, 30)",
                                      "fontSize": 17,
                                      "fontWeight": "600",
                                    }
                                  }
                                >
                                  Journal App
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNCSafeAreaView>
    </View>
  </View>
</View>
`;
